migration/migration_test.go

```diff
@@ -0,0 +1,159 @@
+package migration
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/powergate/tests"
+)
+
+func TestEmptyDatastore(t *testing.T) {
+	t.Parallel()
+
+	ms := map[int]Migration{
+		1: func(_ datastore.Txn) error {
+			return fmt.Errorf("This migration shouldn't be run on empty database")
+		},
+	}
+	m := newMigrator(ms, true)
+
+	v, empty, err := m.getCurrentVersion()
+	require.NoError(t, err)
+	require.Equal(t, 0, v)
+	require.True(t, empty)
+
+	err = m.Ensure()
+	require.NoError(t, err)
+
+	v, _, err = m.getCurrentVersion()
+	require.NoError(t, err)
+	require.Equal(t, 1, v)
+}
+
+func TestNonEmptyDatastore(t *testing.T) {
+	t.Parallel()
+
+	ms := map[int]Migration{
+		1: func(_ datastore.Txn) error {
+			return nil
+		},
+	}
+	m := newMigrator(ms, false)
+
+	v, empty, err := m.getCurrentVersion()
+	require.NoError(t, err)
+	require.Equal(t, 0, v)
+	require.False(t, empty)
+
+	err = m.Ensure()
+	require.NoError(t, err)
+
+	v, _, err = m.getCurrentVersion()
+	require.NoError(t, err)
+	require.Equal(t, 1, v)
+}
+
+func TestNoop(t *testing.T) {
```

Comment:
Test the case of the migration running when the version is already the latest.. so most cases in general when Powergate is restarted or no new migrations exist.